cpp/include/raft/linalg/svd.cuh

```diff
@@ -182,6 +181,218 @@ bool evaluateSVDByL2Norm(const raft::handle_t& handle,
   return detail::evaluateSVDByL2Norm(handle, A_d, U, S_vec, V, n_rows, n_cols, k, tol, stream);
 }
 
+/**
+ * @defgroup svd Singular Value Decomposition
+ * @{
+ */
+
+/**
+ * @brief singular value decomposition (SVD) on a column major
+ * matrix using QR decomposition
+ * @param[in] handle raft::handle_t
+ * @param[in] in input raft::device_matrix_view with layout raft::col_major of shape (M, N)
+ * @param[out] sing_vals singular values raft::device_vector_view of shape (K)
+ * @param[out] left_sing_vecs optional left singular values of raft::device_matrix_view with layout
+ * raft::col_major and dimensions (m, n)
+ * @param[out] right_sing_vecs optional right singular values of raft::device_matrix_view with
+ * layout raft::col_major and dimensions (n, n)
+ */
+template <typename ValueType, typename IndexType>
+void svd_qr(
+  const raft::handle_t& handle,
+  raft::device_matrix_view<ValueType, IndexType, raft::col_major> in,
```

Comment:
Use `const ValueType`. This applies throughout the PR.

Reply:
Many of these svd/eig functions that use cusolver don't have const-qualifers even on the original pointers-based API. I *think* it is because cusolver doesn't accept const-qualified pointers. What to do?

Reply:
Cast the constness away underneath but use it on the outermost bits. This future-proofs the API as well.